src/model.ts

```diff
@@ -81,6 +85,11 @@ import { getXLSX } from "./xlsx/xlsx_writer";
 
 export type Mode = "normal" | "readonly" | "dashboard";
 
+export interface ModelExternalConfig {
```

Comment:
I would move it below `ModelConfig` since `ModelConfig` is more important and more general than `ModelExternalConfig`.
It makes more sense to see `ModelConfig` first